core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -242,6 +244,16 @@ public SystemSessionProperties(
                         "Prefer source table layouts that produce streaming operators",
                         false,
                         false),
+                dataSizeProperty(
+                        MAX_LOCAL_EXCHANGE_BUFFER_SIZE,
+                        "Maximum local exchange buffer size",
+                        taskManagerConfig.getMaxLocalExchangeBufferSize(),
```

Comment:
Totally missed that. Is this necessary to expose the buffer size as a session property? This is potentially dangerous to allow the end users to change that. Imagine a user setting this buffer to something arbitrary large and submitting a query into a shared cluster. Usually we are trying to avoid exposing settings that could impact other sessions to the end users.

Reply:
I'm using this property to test different scenarios in `BaseHiveConnectorTest`. For instance, check `testTaskScaleWritersWhenOnlyLocalExchangeBufferSizeIsSmall`. If we remove this property then testing will become overly complex. Basically separate query runner for each test. Therefore, maybe it's okay to keep this property. In addition, I've made it hidden which might help a little.

Reply:
@raunaqmorarka WDYT?

Reply:
I would prefer to avoid adding this session property, could you write the tests by changing TaskManagerConfig instead ?

Reply:
Changing configs is also not ideal. That would result in a distributed query runner being created 3 times what would make the tests running longer. I wonder if we can choose a slightly different path. The dataset (`orders` table) is fixed. I wonder if we can somehow change the dataset size produced by adjusting the filters while keeping the default buffer size? (maybe take a look how the testing is done for original `scale_writers` at a query level)

Reply:
IMO depending on the filters is also not ideal because we are depending on the default buffer size. What if it changes in the future?

Reply:
I wonder if it would be sufficient to simply test whether scalling is happening or not at the integration test level and provide more fine grained covered in a form of a unit test. What do you think?

Reply:
WDYT? @raunaqmorarka

Reply:
I've removed this property